cpp/arcticdb/toolbox/query_stats.cpp

```diff
@@ -116,37 +136,54 @@ QueryStats::QueryStatsOutput QueryStats::get_stats() const {
     return result;
 }
 
-void QueryStats::add(entity::KeyType key_type, TaskType task_type, uint32_t value) {
+void QueryStats::add(entity::KeyType key_type, TaskType task_type, StatType stat_type, uint32_t value) {
     if (is_enabled()) {
-        stats_by_key_type_[static_cast<size_t>(key_type)][static_cast<size_t>(task_type)].count_.increment(value);
+        auto& stats = stats_by_key_type_[static_cast<size_t>(key_type)][static_cast<size_t>(task_type)];
+        switch (stat_type) {
+            case StatType::TOTAL_TIME_MS:
+                stats.total_time_ns_.increment(value);
+                break;
+            case StatType::COUNT:
+                stats.count_.increment(value);
+                break;
+            case StatType::SIZE_BYTES:
+                stats.size_bytes_.increment(value);
+                break;
+            default:
+                throw std::invalid_argument("Invalid stat type");
```

Comment:
Assertion error with our normal framework to bubble up to the Python layer cleanly?